packages/flutter_tools/lib/src/migrate/migrate_utils.dart

```diff
@@ -0,0 +1,325 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:io';
+import 'dart:typed_data';
+
+import '../base/common.dart';
+import '../base/file_system.dart';
+import '../base/logger.dart';
+
+/// The default name of the migrate working directory used to stage proposed changes.
+const String kDefaultMigrateWorkingDirectoryName = 'migrate_working_dir';
+
+/// Utility class that contains static methods that wrap git and other shell commands.
+class MigrateUtils {
+  MigrateUtils();
+
+  /// Calls `git diff` on two files and returns the diff as a DiffResult.
+  static Future<DiffResult> diffFiles(File one, File two, Logger logger) async {
+    if (one.existsSync() && !two.existsSync()) {
+      return DiffResult.deletion();
+    }
+    if (!one.existsSync() && two.existsSync()) {
+      return DiffResult.addition();
+    }
+    final List<String> cmdArgs = <String>['diff', '--no-index', one.absolute.path, two.absolute.path];
+    final ProcessResult result = await Process.run('git', cmdArgs);
```

Comment:
For testing, don't use `dart:io`'s Process.run, but instead the `ProcessManager` from the context: https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/context_runner.dart#L306. In addition, direct inject the process manager via the constructor, so we can unit test it with a `FakeProcessManager`.

The `IOSDeploy` class is an example that does this.

Reply:
Also, for direct injection, these static methods should probably be moved to instance methods.

Reply:
Ok, thanks for the feedback. I figured I would have to change the static utils class setup, this sounds good, thanks for the example to reference.